src/lighthouse.php

```diff
@@ -355,6 +355,18 @@
 
     'batchload_relations' => true,
 
+    /*
+    |--------------------------------------------------------------------------
+    | Optimized Selects
+    |--------------------------------------------------------------------------
+    |
+    | If set to true, Eloquent will only select the columns necessary to resolve a query.
+    | Use the @select directive to specify column dependencies of compound fields.
```

Comment:
Due to the problems I outlined in `SelectHelper`, I think we need to change the approach to this entirely. I believe we need to require that every field in a model that needs to select any columns has an `@select` configuration. I know that this is cumbersome, but perhaps we can allow iterative adoption by marking some models as being optimizable and only applying the optimization to those.

```graphql
type Foo @select {
  id: ID @select(columns: ["id"])
}
```

`@select` on the model type signifies it can be considered for optimization.